crates/pico_macros/src/memo.rs

```diff
@@ -97,7 +103,9 @@ pub(crate) fn memo(args: TokenStream, item: TokenStream) -> TokenStream {
 
     let mut new_sig = sig.clone();
 
-    if args_.reference {
+    if args_.inner {
+        return_expr = quote! { #return_expr.clone() };
+    } else if args_.inner_ref {
         let lifetime = new_sig.generics.params.iter().find_map(|param| {
             if let GenericParam::Lifetime(lt) = param {
                 Some(&lt.lifetime)
```

Comment:
on line 119 (`ReturnType::Type(parse_quote!(->), Box::new(parse_quote!(&#lt #return_type)));`) do we also need to make sure that that lifetime is attached to the `db` param?

Reply:
hmm, just in case if a function has a lifetime but not the database argument? Ok, let's check that it has the same lifetime as the function.

Reply:
btw, we need a test for it